src/linear_algebra/affine_space/affine_subspace.lean

```diff
@@ -189,6 +189,51 @@ iff.rfl
 
 variables {k P}
 
+/-- Two affine subspaces are equal if they have the same points. -/
+@[ext] lemma coe_injective  : function.injective (coe : affine_subspace k P → set P) :=
+λ s1 s2 h, begin
+  cases s1,
+  cases s2,
+  congr,
+  exact h
+end
+
+@[simp] lemma ext_iff (s₁ s₂ : affine_subspace k P) :
+  (s₁ : set P) = s₂ ↔ s₁ = s₂ :=
+⟨λ h, coe_injective h, by tidy⟩
+
+instance : has_vadd V (affine_subspace k P) :=
+{ vadd := λ x S,
+{ carrier := x +ᵥ S,
```

Comment:
You should add a lemma that says `\u(x +ᵥ S) = x +ᵥ S` and that `(x +ᵥ S).direction = S.direction`